src/services/CompareUtils.jl

```diff
@@ -1,9 +1,40 @@
+## @generated compare
+# Reference https://github.com/JuliaLang/julia/issues/4648
+
+import Base.==
+
+#=
+TODO <:InferenceVariable: please someone confirm this, I'm not fully up to date why softtype was created.
+For now abstract `InferenceVariable`s are considered equal if they are the same type (except for labels, that I feel are deprecated)
+If your implentation has aditional properties such as `DynPose2` with `ut::Int64` (microsecond time) or support different manifolds
+implement compare if needed. softtype sounds a bit if it can be a trait, or in the future, rather a normal julia type.
+=#
+==(a::InferenceVariable,b::InferenceVariable) = typeof(a) == typeof(b)
+
+==(a::ConvolutionObject, b::ConvolutionObject) = typeof(a) == typeof(b)
+
+==(a::FunctorInferenceType, b::FunctorInferenceType) = typeof(a) == typeof(b)
```

Comment:
just checking you know about `===`

Reply:
Jip, but an object returned by a persistent driver will not be `===`